core/controller/src/main/scala/whisk/core/controller/Actions.scala

```diff
@@ -391,7 +394,12 @@ trait WhiskActionsApi
                 checkSequenceActionLimits(entityName, seq.components) map {
                     _ => makeWhiskAction(content.replace(seq), entityName)
                 }
-            case _ => Future successful { makeWhiskAction(content, entityName) }
+            case otherExec =>
+                if (!otherExec.deprecated) {
+                    Future successful makeWhiskAction(content, entityName)
+                } else {
+                    Future failed RejectRequest(BadRequest, runtimeDeprecated(otherExec))
+                }
```

Comment:
It might be more readable to use guards instead of an inner if like:

```
case otherExec if !otherExec.deprecated => Future.successful(makeWhiskAction(content, entityName))
case deprecatedExec => Future.failed(RejectRequest(BadRequest, runtimeDeprecated(otherExec)))
```

Not sure though.